File: week1/Bu$ra/TwoSum.cs
using System.Collections.Generic;

public class Solution {
    public static int[] TwoSum(int[] nums, int target) {
        // Sayıları ve index'lerini tutan bir sözlük oluşturuyoruz.
        Dictionary<int, int> numIndices = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++) {
            int complement = target - nums[i];

            // Eğer complement değeri sözlükte varsa, bu iki sayının toplamı hedefe eşittir.
            if (numIndices.ContainsKey(complement)) {
                return new int[] { numIndices[complement], i };
            }

            // Sözlüğe güncel sayıyı ve index'ini ekliyoruz.
            if (!numIndices.ContainsKey(nums[i])) {
                numIndices[nums[i]] = i;
            }
        }

        // Eğer çözüm bulunamazsa null döndürüyoruz.
        return null;
    }

File: week1/yusuffenes/twoSum.py
def twoSum(self, nums: List[int], target: int) -> List[int]:
    returnedList = []
    for i in range(nums.__len__()-1):
        for x in range(i+1,nums.__len__()):
            if nums[i] + nums[x] == target and returnedList.__len__() == 0:
                returnedList.append(i)
                returnedList.append(x)
    return returnedList